### PR TITLE
Make the backups actually useful

### PR DESCRIPTION
**Pull request by peci1**
In reaction to https://forum.turris.cz/t/restoring-from-a-cloud-backup-experience/8251/4 .

There's a lot of useful stuff that's not currently being backed up. Restoring from such backup after flasing a medkit results in half-working system (especially missing firewall.user).

### CODE REVIEW

**Comment by peci1** [comment]
I added directories for the services I'm currently using. Some of them don't actually have to be there, but I find it nice to have them. The backup scripts should not break if they're missing since there's [`mkdir -p`](https://gitlab.labs.nic.cz/turris/turris-os-packages/blob/a5fc8751972097b76e1d08d76f2747f865df13bd/cznic/turris-maintain/files/config-backup.sh#L41) in them for each of the directories.

Maybe adding a drop-in config directory would also be an option so that application-specific directories don't have all to be listed here... But that's probably a longer-term solution.

**Comment by peci1** [comment]
Ahh, no, the [mkdir -p](https://gitlab.labs.nic.cz/turris/turris-os-packages/blob/a5fc8751972097b76e1d08d76f2747f865df13bd/cznic/turris-maintain/files/config-backup.sh#41) only creates parent directories when given a directory, so if e.g. tor is not installed, putting `/etc/tor` in the backup selection causes the backup to fail.

So, is it possible to change maintain-config-backup so that it doesn't fail on nonexistent directories (but e.g. issues a warning to syslog)?

**Comment by VojtechMyslivec** [comment]
Hello and thanks for the patch, however it is not a good idea to include all of these files into the backup by default.

For start – you have written it yourself – the backups actually fails if you include non-existing directories. A lot of directories you introduce is empty or does not even exists in default setup. And changing the behavior of the backup script is not desirable at the moment.

The backup script is intended to backup Foris and lUCI configuration. Both these tools works only with `/etc/config` directory in most cases. If you fiddle with `/etc/` dir by yourself – which is not used by default – you should include your files in the backup manually. Actually this is what the `/etc/config/backups` is for. I understand the documentation should mention this at the end of *howto* articles.

I agree with `/etc/firewall.user`. This is something that is absolutely should be in the backup (I have include it on my own routers as well).

Passwords and certificates is a question. We prefer to not includes secrets by default in the backups as users are usually not careful about downloaded files. Also, there is another one use case to restore the backup: If you forgot the Foris and/or root password, you can reset the router to factory defaults and restore the config. This will not restore your original passwords so you can use your original config with new password(s).

A bit for the formatting: `uci` tool use tabs by default, please follow these principles. Also mind the trailing spaces.

To sum it up. We will include `/etc/firewall.user` file and also commented-out `/etc/ssl/ca/openvpn`, `/etc/passwd`, `/etc/shadow`, `/root/.ssh` and so for an inspiration.

**Comment by peci1** [comment]
Ok, thank you for considering my PR. Should I edit this PR, or will you instead do the changes yourself and commit them directly?

Don't you think rc.d is also worth considering? I think that the set of autostart applications can be fully controlled via LUCI, and thus it should be included. Of course it might cause problems if an autostarted application is causing problems, but I don't think that would be a frequent case. 

And what about the drop-in config directory for backups? I know it goes against what you stated (that only foris and luci stuff should be backed up), but it could be more comfortable for "half-power-users" who start learning the Linux system inside...

Adding the commented-out inspiration lines is definitely a good idea!

One more thing that came to my mind is that there should be an explanatory section in the cloud backups menu in foris that would explain what exactly is being backed up and it should also mention that any non-foris and non-luci changes have to be added manually to /etc/config/backups. And if you decide openvpn certs and configs should not be a part of the default backup selection, it should be clearly stated on the cloud backup page, so that the user can backup the files manually.

One last thing, more a dream than something I'd believe would really happen - it'd be nice to add a comment to the top of all non-backed-up config files telling the user that he should add the files to backup selection manually if he edits them. This would be exactly one of the things that would make TurrisOS more (power-)user-friendly - in the current state, you have to collect enormous contextual knowledge about the system if you want to change something, because you never know what depends on what and what are the conseqences. Just documenting these consequences **in the place where they "apply"** would be really nice. Having hundreds of pages of HTML documentation is of course important, but being warned at the right place would also be helpful. Or at least adding a comment like "If you are about to change this file, please consult WIKI/wiki/weird/program.html for warnings and tips."

**Comment by VojtechMyslivec** [comment]
> No, I meant a directory like /etc/backups.d, where other programs would install their configs in separate files.

Oh, I see. That's not a bad idea. However packages from upstream are not aware of our backup utility. So we will have to repack every package and that's quite a lot of work...

We are aware that OpenWRT have introduced some backup tool so we should probably migrate to that one in Turris OS 4.0.

If you don't mind, I will close this PR. If you want to discuss about what should be included in the backups by default, don't hesitate to drop a comment in the [gitlab MR](https://gitlab.labs.nic.cz/turris/turris-os-packages/merge_requests/64).
